src/components/authentication-template.ts

```diff
@@ -0,0 +1,214 @@
+import { togglePassword, preventDefault, trimString } from '../services/util';
+
+import '@internetarchive/ia-activity-indicator/ia-activity-indicator';
+
+@customElement('authentication-template')
+export class AuthenticationTemplate extends LitElement {
+  /**
+   * authenticate type eg. ia|third-party
+   *
+   * @type {string}
+   * @memberof AuthenticationTemplate
+   */
+  @property({ type: String }) authenticationType?: string = 'ia';
+
+  /**
+   * user identifier
+   *
+   * @type {string}
+   * @memberof AuthenticationTemplate
+   */
+  @property({ type: String }) identifier: string = '';
+
+  /**
+   * user email/username
+   *
+   * @type {string}
+   * @memberof AuthenticationTemplate
+   */
+  @property({ type: String }) email: string = '';
+
+  /**
+   * show loading indicator when form is submitted
+   *
+   * @private
+   * @type {boolean}
+   * @memberof AuthenticationTemplate
+   */
+  @state() showLoadingIndicator?: boolean;
+
+  /**
+   * hold the password
+   * @private
+   * @type {string}
+   * @memberof IAUXAccountSettings
+   */
+  @state() private password?: string = '';
+
+  /**
+   * hold the password field error
+   * @private
+   * @type {string}
+   * @memberof IAUXAccountSettings
+   */
+  @state() private passwordError?: string = '';
+
+  @query('.password') private passwordField?: HTMLInputElement;
+
+  /**
+   * verify IA password to change you protected settings
+   *
+   * @param {Event} event
+   * @memberof AuthenticationTemplate
+   */
+  async verifyIAPassword(event: Event) {
+    preventDefault(event);
+
+    if (!trimString(this.password as string)) {
+      this.passwordField?.focus();
+      this.passwordError = " * password can't be empty";
+      return;
+    }
+
+    this.showLoadingIndicator = true;
+    const response = (await backendServiceHandler({
+      action: 'verify-password',
+      identifier: this.identifier,
+      password: trimString(this.password as string),
+    })) as ResponseModel;
+
+    if (response.success === true) {
+      // successfully authenticated
+      this.dispatchEvent(new Event('ia-authenticated'));
+    } else {
+      this.passwordField?.focus();
+      this.passwordError = ' * invalid password';
+    }
+
+    this.showLoadingIndicator = false;
+  }
+
+  /**
+   * set password value to property
+   *
+   * @param {Event} e
+   * @memberof AuthenticationTemplate
+   */
+  setPassword(e: Event) {
+    const input = e.target as HTMLInputElement;
+    this.password = input.value;
+    this.passwordError = '';
+  }
+
+  get iaPasswordVerification() {
+    return html`
+      <p>Please verify your password again to change protected settings.</p>
+      <div class="form-element">
+        <label for="password">Internet Archive password</label>
+        <input
+          type="password"
+          class="form-control password"
+          id="password"
+          name="password"
+          required
+          @input=${this.setPassword}
+        />
+        <img
```

Comment:
suggestion: make as button since it's clickable.

general rule, hit area is 30px x 30px